src/core/ddsc/tests/psmxif.c

```diff
@@ -0,0 +1,441 @@
+</Tracing>\
+",
+    cdds_psmx_name,
+    cdds_psmx_name,
+    locator_str,
+    locator ? (int) locator[0] : -1, // This prevents plugins from forwarding across the "network".
+    (int) int_dom // log file name
+  );
+  char *xconfigstr = ddsrt_expand_envvars (configstr, int_dom);
+  ddsrt_free (configstr);
+  return xconfigstr;
+}
+
+static bool endpoint_has_psmx_enabled (dds_entity_t rd_or_wr)
+{
+  dds_return_t rc;
+  struct dds_entity *x;
+  bool psmx_enabled = false;
+  rc = dds_entity_pin (rd_or_wr, &x);
+  CU_ASSERT_FATAL (rc == DDS_RETCODE_OK);
+  switch (dds_entity_kind (x))
+  {
+    case DDS_KIND_READER: {
+      struct dds_reader const * const rd = (struct dds_reader *) x;
+      psmx_enabled = (rd->m_endpoint.psmx_endpoints.length > 0);
+      break;
+    }
+    case DDS_KIND_WRITER: {
+      struct dds_writer const * const wr = (struct dds_writer *) x;
+      psmx_enabled = (wr->m_endpoint.psmx_endpoints.length > 0);
+      break;
+    }
+    default: {
+      CU_ASSERT_FATAL (dds_entity_kind (x) == DDS_KIND_READER || dds_entity_kind (x) == DDS_KIND_WRITER);
+      break;
+    }
+  }
+  dds_entity_unpin (x);
+  return psmx_enabled;
+}
+
+/** @brief Convert a set of stats to a string.
+ * 
+ * Truncates the output string if the string buffer capacity is too small.
+ * 
+ * @param[in] dmock stats to convert
+ * @param[out] str_out string buffer to write the string into
+ * @param[in] str_capacity number of bytes the string buffer can hold
+ * 
+ * @return Upon successful return, it returns the number of characters printed
+ *         (excluding the null byte used to end output to strings).
+ *         If an output error is encountered, a negative value is returned.
+*/
+static int dummy_mockstats_tostring(const dummy_mockstats_t* dmock, char* str_out, size_t str_capacity)
+{
+  return snprintf(
+    str_out,
+    str_capacity,
+    "\
+  create_psmx: %i\n\
+  \n\
+  type_qos_supported: %i\n\
+  create_topic: %i\n\
+  delete_topic: %i\n\
+  deinit: %i\n\
+  get_node_id: %i\n\
+  supported_features: %i\n\
+  \n\
+  create_endpoint: %i\n\
+  delete_endpoint: %i\n\
+  \n\
+  request_loan: %i\n\
+  write: %i\n\
+  take: %i\n\
+  on_data_available: %i\n",
+    dmock->cnt_create_psmx,
+    dmock->cnt_type_qos_supported,
+    dmock->cnt_create_topic,
+    dmock->cnt_delete_topic,
+    dmock->cnt_deinit,
+    dmock->cnt_get_node_id,
+    dmock->cnt_supported_features,
+    dmock->cnt_create_endpoint,
+    dmock->cnt_delete_endpoint,
+    dmock->cnt_request_loan,
+    dmock->cnt_write,
+    dmock->cnt_take,
+    dmock->cnt_on_data_available
+  );
+}
+
+/// @brief Check that creating a domain with more than one psmx interface fails.
+/// @methodology
+/// - Create a config string with two psmx interfaces.
+/// - Try to create a domain using this config string.
+/// - Expectation: Failed to create the domain.
+/// 
+CU_Test(ddsc_psmxif, config_multiple_psmx)
+{
+  dds_domainid_t domainId = 0;
+  const char* cdds_psmx_name1 = "dummy";
```

Comment:
All of this ends up being a constant string. Why not simply write a string constant and pass it to `ddsrt_expand_envvars`?